src/core/service.c

```diff
@@ -1358,10 +1454,39 @@ static int service_collect_fds(
                 rfd_names[n_fds] = NULL;
         }
 
+        if (s->open_files) {
+                size_t n_fds = rn_socket_fds + s->n_fd_store;
+
+                LIST_FOREACH(open_files, f, s->open_files) {
+                        _cleanup_close_ int fd = -1;
+
+                        fd = get_open_file_fd(s, (const OpenFile *) f);
```

Comment:
This cast should not be necessary? Can you include the compilation error you get here?

Reply:
You're right. I's not necessary